lib/src/components/NxCollapsibleItemsSelect/NxCollapsibleMultiSelect/stateful/__tests__/NxStatefulCollapsibleMultiSelect.test.tsx

```diff
@@ -4,93 +4,579 @@
+          expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+
+          fireEvent.mouseOver(trigger);
+          await runTimers();
+
+          const tooltip = screen.getByRole('tooltip');
+
+          expect(tooltip).toHaveTextContent('No options');
+        });
+
+        it('has specified classname when tooltipModifierClass prop is provided', async function() {
+          const trigger = renderAndGetTrigger({
+            disabled: true,
+            disabledTooltip: 'No options',
+            tooltipModifierClass: 'tooltipClass'
+          });
+
+          await runTimers();
+          expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+
+          fireEvent.mouseOver(trigger);
+          await runTimers();
+
+          const tooltip = screen.getByRole('tooltip');
+
+          expect(tooltip).toHaveTextContent('No options');
+          expect(tooltip.querySelector('.tooltipClass')).toBeInTheDocument();
+        });
+
+        it('does not show when disabledTooltip prop is not provided', async function() {
+          const trigger = renderAndGetTrigger({ disabled: true });
+
+          await runTimers();
+          expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+
+          fireEvent.mouseOver(trigger);
+          await runTimers();
+
+          expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+        });
+      });
+    });
   });
 
-  it('collapses items when toggled', function() {
-    const shallowRender = getShallow({
-      isOpen: true
+  describe('filter', function() {
+    const filterView = (props?: Partial<Props>) => quickRender({
+      ...props,
+      filterThreshold: 1
+    });
+
+    it('renders an input with type="text" iff when the number of options ' +
+      'is greater than filterThreshold', function() {
+      expect(quickRender().queryByRole('textbox')).not.toBeInTheDocument();
+      expect(filterView().getByRole('textbox')).toHaveAttribute('type', 'text');
+    });
+
+    it('sets the filter input placeholder as specified by filterPlaceholder, ' +
+      'default is "filter"', function() {
+      const viewWithPlaceholder = filterView({ filterPlaceholder: 'filter cats' });
+
+      expect(filterView().getByRole('textbox')).toHaveAttribute('placeholder', 'filter');
+      expect(viewWithPlaceholder.getByRole('textbox')).toHaveAttribute('placeholder', 'filter cats');
+    });
+
+    it('renders a button with an accessible name of "Clear filter"', async function() {
+      const view = filterView();
+
+      await runTimers();
+      const clearBtn = view.getByRole('button', { name: 'Clear filter' });
+
+      expect(clearBtn).toBeInTheDocument();
+      expect(clearBtn).toHaveAccessibleName('Clear filter');
+    });
+
+    it('clears the input when the clear button is clicked', async function() {
+      const user = userEvent.setup(),
+          view = filterView(),
+          inputEl = view.getByRole('textbox');
+
+      await runTimers();
+
```

Comment:
You should add in the user typing a value into the input, to confirm the clear button clears the input value

Reply:
Thanks for catching it, updated here 4dae2b2d775d4ccb67236a44401f75d0d35f4d91